src/pybind/mgr/alerts/module.py

```diff
@@ -229,12 +234,23 @@ def _send_alert_smtp(self, status, diff):
         self.log.debug('message: %s' % message)
 
         # send
-        if self.smtp_ssl:
-            server = smtplib.SMTP_SSL(self.smtp_host, self.smtp_port)
-        else:
-            server = smtplib.SMTP(self.smtp_host, self.smtp_port)
-        if self.smtp_password:
-            server.login(self.smtp_user, self.smtp_password)
-        server.sendmail(self.smtp_sender, self.smtp_destination, message)
-        server.quit()
-        self.log.debug('Sent email to %s' % self.smtp_destination)
+        try:
+            if self.smtp_ssl:
+                server = smtplib.SMTP_SSL(self.smtp_host, self.smtp_port)
+            else:
+                server = smtplib.SMTP(self.smtp_host, self.smtp_port)
+            if self.smtp_password:
+                server.login(self.smtp_user, self.smtp_password)
+            server.sendmail(self.smtp_sender, self.smtp_destination, message)
+            server.quit()
+            self.log.debug('Sent email to %s' % self.smtp_destination)
+        except smtplib.SMTPException as e:
+            return {
+                'ALERTS_SMTP_ERROR': {
+                    'severity': 'warning',
+                    'summary': 'unable to send alert email',
+                    'count': 1,
+                    'detail': [ str(e) ]
+                }
+            }
```

Comment:
I think this also needs to catch the generic Exception class, to handle connection failures. Here's a presumably nonexhaustive list:

```
server = smtplib.SMTP('name-does-not-resolve', 25)
...
socket.gaierror: [Errno -2] Name or service not known

server = smtplib.SMTP('localhost', 12345)
...
ConnectionRefusedError: [Errno 111] Connection refused

server = smtplib.SMTP('google.com', 25)
...
OSError: [Errno 101] Network is unreachable

server = smtplib.SMTP_SSL('localhost', 25)
...
ssl.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1056)
```